pwa-wp.php

```diff
@@ -41,4 +41,40 @@ function pwawp_init() {
 
 	$wp_web_app_manifest = new WP_Web_App_Manifest();
 	$wp_web_app_manifest->init();
+
+	// These could be in ABSPATH . WPINC . '/script-loader.php' file.
+	/** WordPress Service Workers Class */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/class.wp-service-workers.php';
+
+	/** WordPress Scripts Functions */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/functions.wp-service-workers.php';
+
+	add_action( 'wp_print_footer_scripts', 'wp_print_service_workers' );
+
+	// Alternative for this could be in wp-includes/functions.php.
+	add_action( 'template_redirect', 'pwawp_maybe_display_sw_script' );
+
+	add_action( 'init', 'pwawp_add_sw_rewrite_rules' );
+}
+
+/**
+ * Register rewrite rules for Service Workers.
+ */
+function pwawp_add_sw_rewrite_rules() {
+	add_rewrite_tag( '%wp_service_worker%', '(0|1)' );
+	add_rewrite_tag( '%scope%', '([^&]+)' );
```

Comment:
What do you think about combining these two tags? Since the `wp_service_worker` would always get the value of `1`, could we not let the `scope` be the value of that query var?

Reply:
To confirm if I understand correctly:
Do you mean that we could ignore `wp_service_worker` as a separate query var since it's always `1` and just use `scope`, for example by using the following rewrite rule instead and just adding a rewrite tag for `scope`?
`add_rewrite_rule( '^wp-service-worker.js\\?scope=([^&]+)?', 'index.php?scope=$matches[1]', 'top' );`

Added the `wp_service_worker` tag separately since in case the pretty permalinks aren't enabled then we'd need to check the `wp_service_worker=1` separately, otherwise the logic will always run if `$_GET['scope']` is set, even if the `wp_service_worker` is not.

Thoughts?

(Likely that you meant something else by the comment.)

Reply:
Perhaps you meant the opposite -- that we could just use `wp_service_worker`, for example:
`add_rewrite_tag( '%wp_service_worker%', '([^&]+)' );` where scope is the value of `wp_service_worker` and use 
`add_rewrite_rule( '^wp-service-worker.js\\?scope=([^&]+)?', 'index.php?wp_service_worker=$matches[1]', 'top' );`?

In this case as well in case of non-pretty permalinks this wouldn't work with the current registering URL structure, unless we would use the return URL as `?wp_service_worker=$scope` for the return URL in case of non-pretty permalinks. Thoughts?

Reply:
Yes, I meant the latter, just using the one `wp_service_worker` query var. If pretty permalinks are disabled, then the URL would be `/?wp_service_worker=$scope`.